src/service/plugins/sftp.js

```diff
@@ -317,6 +318,54 @@ var Plugin = GObject.registerClass({
         }
     }
 
+    /**
+     * Create a symbolic link referring to the device by name
+     */
+    _addSymlink(mount) {
+        let by_name_dir = Gio.File.new_for_path(
+            gsconnect.runtimedir + '/by-name/'
+        );
+        try {
+            by_name_dir.make_directory_with_parents(null);
+        } catch (e) {
+            if ( ! e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.EXISTS) ) {
+                throw e;
```

Comment:
Since this function is not critical to the mount succeeding, it's probably best not to re-throw errors in this function. Instead of `throw e;` just use:

```js
debug(e, `${this.device.name}: ${this.name}`);
```
...or `warning()` if you think it's an error a user might want to have in the log.